packages/embark-mocha-tests/src/lib/index.js

```diff
@@ -35,19 +47,96 @@ class MochaTestRunner {
     return JAVASCRIPT_TEST_MATCH.test(path);
   }
 
-  run(options, cb) {
+  async checkDeploymentOptions(options) {
```

Comment:
this feels out of place to me but I can't quite put my finger on it. it feels somewhat off that the mocha tests plugin would be starting & stopping the blockchain node.I wonder if there is a different way in which we could pass this responsibility to the test stack component, for e.g the mocha tests plugin could pass the intended config, and the test stack component could then do whatever it's neeeded.

Reply:
You're right. Here is the update: https://github.com/embark-framework/embark/pull/1870/commits/fd9cd165cbc84021493267ecc4b3ea3044f572e0